### PR TITLE
Lpfg 520 manage learner provider

### PR DESCRIPTION
**Pull request by mick-jones**
- Added learningProviderController.js, learningProviderFactory.js
- Validation on input for adding a learner provider
- Added add-terms-and-conditions page
- Removed goBack.js
- Styling added for provider list; new template for listing learning providers
- Refactored backbutton & phaseBanner
- Changed the way right hand menu works
- Nav links added to main template and removed from individual pages

### CODE REVIEW

File: .prettierrc.json [hunk 1]
```diff
@@ -5,5 +5,6 @@
 	"singleQuote": true,
 	"trailingComma": "es5",
 	"useTabs": true,
-	"tabWidth": 4
+	"tabWidth": 4,
+	"printWidth": 120
```
Comment (m24murray) [inline .prettierrc.json]: setting as 120 chars as prettier is make some _pretty_ awful formatting decisions. But happy to remove if anyone feels strongly

File: src/controllers/courseController.ts [hunk 2]
```diff
@@ -109,4 +99,28 @@ export class CourseController {
...
+	public getAddLearningProviderToCourse() {
+		return async (request: Request, response: Response) => {
+			response.render('page/add-learning-provider-to-course')
+		}
+	}
+
+	public setAddLearningProviderToCourse() {
+		return async (request: Request, response: Response) => {
+			response.render('page/add-learning-provider-to-course')
+		}
```
Comment (m24murray) [inline src/controllers/courseController.ts]: Not actually using these methods right now, but will be used when finish up the T&Cs and Cancellation Policies

File: views/component/providerList/template.njk [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<h1 class="govuk-heading-l">Learning providers</h1>
+<ul class="list">
+    <li class="provider-headers"><span class="govuk-body first-header">Learning provider name</span><span class="govuk-body">Added on</span></li>
+    {% for learningProvider in pageResults.results %}
+
+    <li class="list-details plist-details u--border--bgrey">
+        <p class="govuk-body list-item list-name" href="/content-management/course/{{ learningProvider.id }}">{{ learningProvider.name }}</p>
+        <p class="govuk-body list-item list-date">24th Jan 2018</p>
+        <a class="govuk-link list-link " href="#">Edit</a>
```
Comment (m24murray) [inline views/component/providerList/template.njk]: Just hardcoding to # until we get edit functionality up and running

File: views/page/add-learning-provider-to-course.html [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+{% block content %}   
+    <div class="govuk-grid-row">
+        <div class="govuk-grid-column-two-thirds">
+            <h3 class="govuk-body form-header govuk-!-font-size-24">Add a new course</h3>
+            <h1 class="govuk-heading-m govuk-!-font-size-36">Learning Providers</h1>
+
+            <form action="/form-hander" method="post">
+                <fieldset class="govuk-fieldset select-with-button">
+                    {{ govukSelect({
```
Comment (m24murray) [inline views/page/add-learning-provider-to-course.html]: Hard coding for now until learning catalogue changes come through

File: .prettierrc.json (hunk 1 above)
Comment (elxsy) [reply .prettierrc.json]: We have big screens, set it to something longer, please

File: src/learning-catalogue/learning-provider/index.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import {AxiosInstance} from 'axios'
+import {LearningCatalogueConfig} from '../learningCatalogueConfig'
+import {DefaultPageResults} from '../model/defaultPageResults'
+import {LearningProvider} from '../model/learningProvider'
+import {RestService} from '../service/restService'
+import {LearningProviderService} from '../service/learningProviderService'
+
+export class LearningProviderCatalogue {
```
Comment (elxsy) [inline src/learning-catalogue/learning-provider/index.ts]: Manager more suitable name?

File: .prettierrc.json (hunk 1 above)
Comment (m24murray) [reply .prettierrc.json]: Tbh, i think 120 is a good compromise, especially for viewing on things like Github. Can we leave it at 120 and increase if necessary?

File: src/applicationContext.ts [hunk 6]
```diff
@@ -13,22 +13,30 @@ import {CourseValidator} from './learning-catalogue/validator/courseValidator'
 import {EnvValue} from 'ts-json-properties'
 import {CourseController} from './controllers/courseController'
 import {CourseFactory} from './learning-catalogue/model/factory/courseFactory'
+import {LearningProviderController} from './controllers/learningProviderController'
```
[the remark this answers is not in the document]
Comment (m24murray) [reply src/applicationContext.ts]: Agreed, but can we do this in a future PR ?

File: src/controllers/learningProviderController.ts [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+		this.learningProvider = learningProvider
+		this.learningProviderValidator = learningProviderValidator
+		this.learningProviderFactory = learningProviderFactory
+	}
+
+	public index() {
+		logger.debug('Getting Learning Providers')
+		const self = this
+
+		return async (request: Request, response: Response) => {
```
[the remark this answers is not in the document]
Comment (m24murray) [reply src/controllers/learningProviderController.ts]: Done

**Comment by elxsy** [comment]
Next work, package structure to be changed

**Pull request merged**